hackIDE/static/hackIDE/js/custom.js

```diff
@@ -210,7 +210,13 @@ $(document).ready(function(){
 		var zip = new JSZip()
 		zip.file(filename+"."+ext, text)
 		var downloaded = zip.generate({type : "blob"})
-		saveAs(downloaded, "test.zip")
+
+		var user_filename_choice = prompt("Enter a filename for the .zip", filename)
+		if (user_filename_choice == null || user_filename_choice == "" || user_filename_choice == " " ){
```

Comment:
Why are you checking for `" "` here? Why not for `"  "` (2 or more spaces) as well?

I'd say a better approach will be to strip all leading/trailing spaces and check if there's any character left.

Also, I'd like to see a variable (may be `filename`) being set in if/else statements and that being used below for `saveAs` command. Basically, I'd like to see `saveAs` only once because that will make it easy to make any other logical changes in future. What's your opinion about this?

Nit: Indentation inside if/else statements also needs to be fixed.